controllers/machineset_controller.go

```diff
@@ -547,53 +547,32 @@ func (r *MachineSetReconciler) MachineToMachineSets(o client.Object) []ctrl.Requ
 	return result
 }
 
-func (r *MachineSetReconciler) getMachineSetsForMachine(ctx context.Context, m *clusterv1.Machine) []*clusterv1.MachineSet {
+func (r *MachineSetReconciler) getMachineSetsForMachine(ctx context.Context, m *clusterv1.Machine) ([]*clusterv1.MachineSet, error) {
 	log := ctrl.LoggerFrom(ctx, "machine", m.Name)
 
 	if len(m.Labels) == 0 {
 		log.Info("No machine sets found because it has no labels")
-		return nil
+		return nil, nil
 	}
 
 	msList := &clusterv1.MachineSetList{}
-	err := r.Client.List(ctx, msList, client.InNamespace(m.Namespace))
-	if err != nil {
-		log.Error(err, "Failed to list machine sets")
-		return nil
+	if err := r.Client.List(ctx, msList, client.InNamespace(m.Namespace)); err != nil {
+		return nil, fmt.Errorf("failed to list MachineSets: %w", err)
```

Comment:
nit: Maybe also re-introduce the logger here.
```suggestion
              log.Error(err, "Failed to list machine sets")
		return nil, errors.Wrapf(err, "failed to list MachineSets")
```

The second one is more of a general thing, in most of the places in the codebase we use `Wrapf` instead of `Errorf`. Maybe we should use a single practice across (with the emphasis on **maybe**)

Reply:
`errors.Wrapf` is around probably only because there was not native support for it before golang 1.13. Unless there's a reason to not I suggest we rely on the standard errors library package and follow `fmt.Errorf %w", err)` pattern for any upcoming changes rather than perpetuate the old pattern.

Reply:
I agree on the `fmt.Errorf`, but I like the suggestion to keep the logging.

Reply:
This function is now returning an appropriate concatenated error string. It's a consumer of the function choice to handle the error, which in this case is already logging, so if we log here it'd be duplicated https://github.com/kubernetes-sigs/cluster-api/pull/4652/files#diff-2678c5ef8cd150e20f9ea965dcaaea54e17e2c8010eb37447b5745760aca890fR538-R540

Reply:
We usually use the pkg/errors package instead of fmt

Reply:
Updated anyways so we can move along.

Reply:
pkg/errors is in general easier to use, and it's used pretty much everywhere within k8s https://grep.app/search?q=pkg/errors&filter[repo][0]=kubernetes/kubernetes — In general though, if it's not broken why switch?